### DISCUSSION TITLE
Can `Raft` nodes be reused after a `shutdown().await`?

### DISCUSSION THREAD

**Discussion by semirix** [databendlabs/openraft#318] [Q&A] [answered]
If my cluster encounters an error and needs to shutdown, can I reuse the `Raft` object I created using `Arc<Store>` and `Arc<Network>` by calling initialize on it again? At the moment, I just reset the network and store objects and create a whole new instance but I was wondering if there was a better way?

**Comment by drmingdrmer** [comment, marked as answer]
Reusing the struct `Raft` would be impossible: It creates several channels when initializing. One of the receiving end or sending end of these channels will be dropped after `shutdown()`.

Reusing `Arc<Store>` or `Arc<Network>` for a new `Raft` struct would be OK if, in your implementation, there are not any states in the `store` or `network` whose validity depends on a `Raft` instance.

**Reply by semirix** [reply]
There are no dependent states. I have special methods on `Arc<Store>` and `Arc<Network>` to reset them if the node shuts down to avoid any corruption.

**Reply by drmingdrmer** [reply]
Then it would be all right reusing them:)
